### PR TITLE
Update pbiviz-tools - Improve new visual template in pbiviz tool

### PR DESCRIPTION
**Pull request by vtkalek**

### CODE REVIEW

File: templates/visuals/default/src/settings.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+
+module powerbi.extensibility.visual {
+    import DataViewObjectsParser = powerbi.extensibility.utils.dataview.DataViewObjectsParser;
+
+    export class VisualSettings extends DataViewObjectsParser {
+    	// TODO: fill all visual settings here
+        // public dataPoint: DataPointSettings = new DataPointSettings();
```
Comment (AviSander) [inline templates/visuals/default/src/settings.ts]: nit: indentation

File: templates/package.json.template [hunk 2]
```diff
@@ -1,6 +1,9 @@
 {
     "version": <%= JSON.stringify(visualData.version) %>,
     "author": <%= JSON.stringify(authorData) %>,
+	"devDependencies": {
+    "powerbi-visuals-utils-dataviewutils": "1.0.1"
```
Comment (ignatvilesov) [inline templates/package.json.template]: Indentation. Please format the file.

File: templates/pbiviz.json.template [hunk 3]
```diff
@@ -14,10 +14,12 @@
...
-    "assets": {
-        "icon": "assets/icon.png"
-    },
-    "externalJS": [],
+  "assets": {
+    "icon": "assets/icon.png"
+  },
+  "externalJS": [
+    "node_modules/powerbi-visuals-utils-dataviewutils/lib/index.js"
+  ],
```
Comment (ignatvilesov) [inline templates/pbiviz.json.template]: Indentation. Please format the file.

File: templates/pbiviz.json.template [hunk 4]
```diff
@@ -14,10 +14,12 @@
         "name": "",
         "email": ""
     },
-    "assets": {
-        "icon": "assets/icon.png"
-    },
-    "externalJS": [],
+  "assets": {
+    "icon": "assets/icon.png"
```
Comment (ignatvilesov) [inline templates/pbiviz.json.template]: Indentation. Please format the file.

File: templates/visuals/default/src/settings.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+	// TODO: fill all visual settings here
+    // export class DataPointSettings {
```
Comment (ignatvilesov) [inline templates/visuals/default/src/settings.ts]: Indentation. Please format the file.

File: templates/visuals/default/src/visual.ts [hunk 6]
```diff
@@ -39,5 +39,9 @@ module powerbi.extensibility.visual {
             console.log('Visual update', options);
             this.target.innerHTML = `<p>Update count: <em>${(this.updateCount++)}</em></p>`;
         }
+
+        public enumerateObjectInstances(options: EnumerateVisualObjectInstancesOptions): VisualObjectInstance[] | VisualObjectInstanceEnumerationObject {
+            return VisualSettings.enumerateObjectInstances(VisualSettings.getDefault(), options);
```
Comment (ignatvilesov) [inline templates/visuals/default/src/visual.ts]: You shouldn't use ```VisualSettings.getDefault()```.
You have to do the following:
* Use ```VisualSettings.parse```
* Save result of the parsing to a variable of the class
* Use the variable instead  ```getDefault```

File: templates/visuals/rvisual/src/settings.ts [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+    export class VisualSettings extends DataViewObjectsParser {
+    	// TODO: fill all visual settings here
+        // public dataPoint: DataPointSettings = new DataPointSettings();
+    }
+
+	// TODO: fill all visual settings here
+    // export class DataPointSettings {
+    //    public fill: string = "#005c55";
+    // }
```
Comment (ignatvilesov) [inline templates/visuals/rvisual/src/settings.ts]: Indentation. Please format the file.

File: templates/visuals/rvisual/src/visual.ts [hunk 8]
```diff
@@ -114,23 +114,8 @@ module powerbi.extensibility.visual {
...
-                            lineColor: this.settings.lineColor,
-                         },
-                        selector: null
-                    });
-                    break;
-            };*/
-
-            return objectEnumeration;
+        public enumerateObjectInstances(options: EnumerateVisualObjectInstancesOptions): VisualObjectInstance[] | VisualObjectInstanceEnumerationObject {
+            return VisualSettings.enumerateObjectInstances(VisualSettings.getDefault(), options);
```
Comment (ignatvilesov) [inline templates/visuals/rvisual/src/visual.ts]: I've left the comment above, please take a look at the ```templates/visuals/default/src/visual.ts```

File: templates/visuals/table/src/settings.ts [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    import DataViewObjectsParser = powerbi.extensibility.utils.dataview.DataViewObjectsParser;
+
+    export class VisualSettings extends DataViewObjectsParser {
+    	// TODO: fill all visual settings here
+        // public dataPoint: DataPointSettings = new DataPointSettings();
+    }
+
+	// TODO: fill all visual settings here
+    // export class DataPointSettings {
+    //    public fill: string = "#005c55";
```
Comment (ignatvilesov) [inline templates/visuals/table/src/settings.ts]: Indentation. Please format the file.

File: templates/visuals/table/src/visual.ts [hunk 10]
```diff
@@ -84,5 +84,9 @@ module powerbi.extensibility.visual {
             let prefix = d3.formatPrefix(d);
             return d3.round(prefix.scale(d), 2) + ' ' + prefix.symbol
         }
+
+        public enumerateObjectInstances(options: EnumerateVisualObjectInstancesOptions): VisualObjectInstance[] | VisualObjectInstanceEnumerationObject {
+            return VisualSettings.enumerateObjectInstances(VisualSettings.getDefault(), options);
+        }
```
Comment (ignatvilesov) [inline templates/visuals/table/src/visual.ts]: The same. Please see above.

**Pull request closed without merging**